numberOfProvinces.js

```diff
@@ -0,0 +1,106 @@
+/*
+n cities
+
+a - b - c
+
+i: n x n matrix
+o: total number of provinces
+
+directly or indirectly
+
+Input: isConnected = [[1,1,0],[1,1,0],[0,0,1]]
+Output: 2
+
+   0 1 2
+0 [1,1,0],
+1 [1,1,0],
+2 [0,0,1]
+
+*/
+
+const { Queue } = require('../utils/queue');
+
+const directions = [
+    [1, 0],
+    [0, 1],
+    [-1, 0],
+    [0, -1],
+];
+
+const CONNECTED = 1;
+
+const getRowColString = (row, col) => `${row}, ${col}`;
+
+const isBound = (row, col, matrix) => {
+    const rowBound = matrix.length;
+    const colBound = matrix[0].length;
+
+    const isRowBound = row >= 0 && row < rowBound;
+    const isColBound = col >= 0 && col < colBound;
+
+    return isRowBound && isColBound;
+};
+
+const getChildren = (row, col, matrix) => {
+    const children = [];
+
+    for (const direction of directions) {
+        const [rowDir, colDir] = direction;
```

Comment:
`rowDir` is not a clear name. I personally read that as rowDirection, so I'm expecting it to somehow indicate the direction (e.g. "DOWN"). Instead, this variable represents the _change_ that we want to make to our row, so I'd name it something like `rowChange`, `deltaRow`, `rowShift`, etc.